File: src/main/java/team4/footwithme/chat/api/ChatroomApi.java
package team4.footwithme.chat.api;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;
import team4.footwithme.chat.api.request.ChatroomRequest;
import team4.footwithme.chat.service.ChatroomService;
import team4.footwithme.chat.service.response.ChatroomResponse;
import team4.footwithme.global.api.ApiResponse;
import team4.footwithme.member.jwt.PrincipalDetails;

import java.util.List;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/v1/chat/room")
public class ChatroomApi {
    private final ChatroomService chatroomService;

    /**
     * 채팅방 생성
     * 팀, 예약 생성시 같이 실행해주기
     */
    @PostMapping
    public ApiResponse<ChatroomResponse> createChatroom(@RequestBody @Valid ChatroomRequest chatroomRequest) {
        return ApiResponse.created(chatroomService.createChatroom(chatroomRequest.toServiceRequest()));
    }

    /**
     * 채팅방 삭제
     */
    @DeleteMapping("/{chatroomId}")
    public ApiResponse<Long> deleteChatroom(@PathVariable("chatroomId") Long chatroomId) {
        return ApiResponse.ok(chatroomService.deleteChatroomByChatroomId(chatroomId));
    }

    /**
     * 채팅방 수정
     */
    @PutMapping("/{chatroomId}")
    public ApiResponse<ChatroomResponse> updateChatroom(@PathVariable("chatroomId") Long chatroomId, @RequestBody @Valid ChatroomRequest chatroomRequest) {
        return ApiResponse.ok(chatroomService.updateChatroom(chatroomId, chatroomRequest.toServiceRequest()));
    }

    /**
     * 채팅방 조회
     */
    @GetMapping
    public ApiResponse<List<ChatroomResponse>> getMyChatroomList(@AuthenticationPrincipal PrincipalDetails principalDetails) {
        return ApiResponse.ok(chatroomService.getMyChatroom(principalDetails.getMember()));
    }
}

File: src/main/java/team4/footwithme/chat/api/ChatroomApi.kt
package team4.footwithme.chat.api

import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*
import team4.footwithme.chat.api.request.ChatroomRequest
import team4.footwithme.chat.service.ChatroomService
import team4.footwithme.chat.service.response.ChatroomResponse
import team4.footwithme.global.api.ApiResponse

@RestController
@RequestMapping("/api/v1/chat/room")
class ChatroomApi(private val chatroomService: ChatroomService) {
    /**
     * 채팅방 생성
     * 팀, 예약 생성시 같이 실행해주기
     */
    @PostMapping
    fun createChatroom(@RequestBody chatroomRequest: @Valid ChatroomRequest?): ApiResponse<ChatroomResponse?> {
        return ApiResponse.Companion.created<ChatroomResponse?>(chatroomService.createChatroom(chatroomRequest!!.toServiceRequest()))
    }

    /**
     * 채팅방 삭제
     */
    @DeleteMapping("/{chatroomId}")
    fun deleteChatroom(@PathVariable("chatroomId") chatroomId: Long?): ApiResponse<Long?> {
        return ApiResponse.Companion.ok<Long?>(chatroomService.deleteChatroomByChatroomId(chatroomId))
    }

    /**
     * 채팅방 수정
     */
    @PutMapping("/{chatroomId}")
    fun updateChatroom(
        @PathVariable("chatroomId") chatroomId: Long?,
        @RequestBody chatroomRequest: @Valid ChatroomRequest?
    ): ApiResponse<ChatroomResponse?> {
        return ApiResponse.Companion.ok<ChatroomResponse?>(
            chatroomService.updateChatroom(
                chatroomId,
                chatroomRequest!!.toServiceRequest()
            )
        )
    }
}
